### PR TITLE
feat: replace sendTransaction with Transaction class

### PR DESCRIPTION
**Pull request by RauppRafael**

### CODE REVIEW

File: src/classes/Transaction.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+export class Transaction {
+    /**
+     * Constructs a new Transaction instance.
+     *
+     * @param transaction - PreparedTransactionRequest object containing transaction details.
+     * @param signer - Signer object to sign and send the transaction.
+     */
+    public constructor(
+        public readonly transaction: PreparedTransactionRequest,
```
Comment (iankressin) [inline src/classes/Transaction.ts]: I'd suggest to rename this to `preparedTransactionRequest`, to match the interface that this parameter is representing.

Or at least use a different name than it's being used to declare the class. Semantically speaking is strange that the `Transaction` receives a `transaction`

File: src/classes/Ethereum.ts [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
+     *                        The value is represented in thousandths. A value of 1_000 denotes no multiplier
+     *                        (i.e., actual gas limit), while a value of 2_000 means the gas limit is doubled.
+     *                        If not provided, a default value of 2_000 is used.
+     */
     constructor({
         defaultChainId,
         availableChainIds,
         chainToRpcMap,
         onWalletUpdate,
+        gasMultiplier = 2_000n, // multiplied by 1_000
```
Comment (iankressin) [inline src/classes/Ethereum.ts]: Is this comment correct?

Given the description above, `gasMultiplier` is already passed in thousands, it isn't multiplied by 1000

**Review by iankressin — CHANGES_REQUESTED** [review]

File: src/classes/Transaction.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+     *
+     * @param transaction - PreparedTransactionRequest object containing transaction details.
+     * @param signer - Signer object used to estimate gas and sign the transaction.
+     * @param gasMultiplier - A factor used to adjust the estimated gas limit for the transaction.
+     *                        The value is represented in thousandths, where 1_000 means no adjustment
+     *                        (i.e., actual estimated gas limit), and 2_000 doubles the gas limit.
+     * @returns - A promise that resolves to a new Transaction instance.
+     */
+    public static async initialize(
+        transaction: PreparedTransactionRequest,
```
Comment (iankressin) [inline src/classes/Transaction.ts]:
```suggestion
        preparedTransactionRequest: PreparedTransactionRequest,
```

File: src/classes/Transaction.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+     *
+     * @param transaction - PreparedTransactionRequest object containing transaction details.
+     * @param provider - Provider object used to estimate the gas limit.
+     * @param gasMultiplier - A factor used to adjust the estimated gas limit for the transaction.
+     *                        The value is represented in thousandths, where 1_000 means no adjustment
+     *                        (i.e., actual estimated gas limit), and 2_000 doubles the gas limit.
+     * @returns - A promise that resolves to the calculated gas limit.
+     */
+    public static async estimateGas(
+        transaction: PreparedTransactionRequest,
```
Comment (iankressin) [inline src/classes/Transaction.ts]:
```suggestion
        preparedTransactionRequest: PreparedTransactionRequest,
```

File: src/classes/Ethereum.ts [hunk 5]
```diff
@@ -39,6 +56,32 @@ export class Ethereum {
...
+     * The gas limit is estimated using the `gasMultiplier` which affects the final gas cost of the transaction.
+     *
+     * @param transaction - The transaction details to initialize.
+     * @param gasMultiplier - Optional. A factor used to adjust the estimated gas limit for the transaction.
+     *                        The value is represented in thousandths, where 1_000 means no adjustment
+     *                        (i.e., actual estimated gas limit), and 2_000 doubles the gas limit.
+     *                        If not provided, the Ethereum instance's default gasMultiplier is used.
+     */
+    public async initializeTransaction(
+        transaction: PreparedTransactionRequest,
```
Comment (iankressin) [inline src/classes/Ethereum.ts]:
```suggestion
        preparedTransactionRequest: PreparedTransactionRequest,
```

File: src/classes/Ethereum.ts [hunk 6]
```diff
@@ -39,6 +56,32 @@ export class Ethereum {
...
+        gasMultiplier?: bigint,
+    ): Promise<Transaction> {
+        return Transaction.initialize(
+            transaction,
```
Comment (iankressin) [inline src/classes/Ethereum.ts]:
```suggestion
            preparedTransactionRequest,
```

**Review by iankressin — CHANGES_REQUESTED** [review]

**Review by iankressin — APPROVED** [review]

**Review by matheus-pissaia — APPROVED** [review]

**Pull request merged**